File: Python/BinarySearch.py
def binarySearch(list, item):
	low = 0
	high = len(list) - 1

	while low <= high:
		mid = (low + high) // 2
		answer = list[mid]
		if answer == item:
			return item
		if answer > item:
			high = mid -1
		elif answer < item:
			low = mid + 1
	return None

myList = [1, 3, 5,8,9,10,15,16,19,26,35]

binarySearch(myList, 15)

File: C++/BinarySearch.cpp
#include <iostream>
#include <vector>

int main()
{
	int target = 50;
	std::vector<int> array(1000001);

	for (int i = 0; i <= 1000000; i++)
		array[i] = i;

	int *low = &array[0];
	int *high = &array[array.size() - 1];

	while (low <= high)
	{
		int *mid = low + (high - low) / 2;
		int anw = *mid;

		if (anw == target)
			return 0;
		if (anw > target)
			high = mid - 1;
		else
			low = mid + 1;
	}

	return -1;
}